### PR TITLE
Problem 4:

### PR DESCRIPTION
**Pull request by rrugamba**
1. Test file (testing.cs)
2. Class Definition (Number(1).cs)
3. Class main (Program(2).cs)

### CODE REVIEW

File: Testing.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            Assert.AreEqual(990009, number.ProductOfTwo3DigitNumbers(num1, num2));
+        }
+
+        [TestMethod]
+        public void CheckIfProductIsPalindromic()
+        {
+            int num1 = 91;
+            int num2 = 99;
+            long pdt = number.ProductOfTwo2DigitNumbers(num1, num2);
+            Assert.AreEqual(true, number.CheckPalindromic(pdt));
```
Comment (dougbradbury) [inline Testing.cs]: There is no negative test case, which tells me that this solution violated rule #3: Only write enough production code as is sufficient to pass the failing test.  This test could be passed by returning a hard-coded true.

File: Testing.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            int num2 = 99;
+            Assert.AreEqual(9009, number.ProductOfTwo2DigitNumbers(num1, num2));
+        }
+
+        [TestMethod]
+        public void ReturnsCorrectProductOfTwo3DigitNumbers()
+        {
+            int num1 = 991;
+            int num2 = 999;
+            Assert.AreEqual(990009, number.ProductOfTwo3DigitNumbers(num1, num2));
```
Comment (dougbradbury) [inline Testing.cs]: These two functions are the same and have nothing to do with how many digits are in the number.

File: Number (1).cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        public bool CheckPalindromic(long number)
+        {
+            string str = number.ToString();
+            int lengthOfnum = str.Count();
+            int itrBegin = 0;
+            int itrEnd = lengthOfnum-1;
+            if (lengthOfnum % 2 == 0)
+            {
+                //test palindromic number (even length)[910019]
```
Comment (dougbradbury) [inline Number (1).cs]: A comment like this is a good indication that a function should be extracted.

File: Number (1).cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                while (str[itrBegin] == str[itrEnd])
+                {
+                    if (itrBegin == ((lengthOfnum / 2) - 1) && itrEnd == (lengthOfnum / 2))
+                        return true;
```
Comment (dougbradbury) [inline Number (1).cs]: This function gets really deeply nested. Another sign that a function should be extracted.

File: Testing.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+namespace TestingPilandromic
+{
+    [TestClass]
+    public class Testing
+    {
+        private Number number;
+        [TestInitialize]
+        public void Iniatilize()
+        {
+            number = new Number();
```
Comment (dougbradbury) [inline Testing.cs]: Number is not a very meaningful class name.

File: Number (1).cs (hunk 3 above)
Comment (rrugamba) [reply Number (1).cs]: I added new files that have more functions, addressing all the reviews

File: Number (1).cs (hunk 4 above)
Comment (rrugamba) [reply Number (1).cs]: added new function that breaks the nested-ness in my previous funciton:
new file has the changes

**Comment by rrugamba** [comment]
Yes i changed and started from scratch, just forgot to delete the old files that are a bit confusing!

just deleted them. doublel check on the new files.

**Pull request closed without merging**